src/SLCore.IntegrationTests/FileAnalysisTestsRunner.cs

```diff
@@ -141,22 +154,34 @@ public async Task VerifyAnalysisSkipped(
         }
     }
 
-    private async Task<(Guid analysisId, TaskCompletionSource<RaiseFindingParams<RaisedIssueDto>> analysisRaisedIssues)> SetUpAnalysis(ITestingFile testingFile, string configScope, bool sendContent)
+    private async Task<TaskCompletionSource<RaiseFindingParams<RaisedIssueDto>>> SetUpAnalysis(
+        ITestingFile testingFile,
+        string configScope,
+        bool sendContent,
+        string compilationDatabasePath = null)
     {
         SetUpListFiles(testingFile.RelativePath, sendContent, configScope, testingFile.GetFullPath());
-        var analysisId = Guid.NewGuid();
         var analysisReadyCompletionSource = new TaskCompletionSource<DidChangeAnalysisReadinessParams>();
         var analysisRaisedIssues = new TaskCompletionSource<RaiseFindingParams<RaisedIssueDto>>();
         SetUpAnalysisListener(
             configScope,
-            analysisId,
             analysisReadyCompletionSource,
             analysisRaisedIssues);
-
         activeConfigScopeTracker.SetCurrentConfigScope(configScope);
+        SetupCompilationDatabase(configScope, compilationDatabasePath);
 
-        await ConcurrencyTestHelper.WaitForTaskWithTimeout(analysisReadyCompletionSource.Task, "analysis readiness");
-        return (analysisId, analysisRaisedIssues);
+        await ConcurrencyTestHelper.WaitForTaskWithTimeout(analysisReadyCompletionSource.Task, "analysis readiness", AnalysisReadinessWaitTimeout);
+        return analysisRaisedIssues;
+    }
+
+    private void SetupCompilationDatabase(string configScope, string compilationDatabasePath)
+    {
+        if (compilationDatabasePath is null)
+        {
+            return;
+        }
+        slCoreTestRunner.SLCoreServiceProvider.TryGetTransientService(out ICFamilyAnalysisConfigurationSLCoreService slCoreService).Should().BeTrue();
+        slCoreService.DidChangePathToCompileCommands(new DidChangePathToCompileCommandsParams(configScope, compilationDatabasePath));
```

Comment:
This is definitely better than it was originally, however I'm still not a fan of the 'short time out to verify it's not called' because it may be too short in some scenarios. Is there no way to receive an event from SLCore that analysis processing has been completed? For example, via reportProgress methods?